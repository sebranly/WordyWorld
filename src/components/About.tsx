// Vendor
import * as React from "react";
import {
  Body,
  Container,
  Content,
  Icon,
  Left,
  List,
  ListItem,
  Separator,
  Text,
  Thumbnail
} from "native-base";
import { Linking, StyleSheet } from "react-native";

// Internal
import { IS_TEST } from "../config/settings";

export interface AboutProps {}

const About: React.FC<AboutProps> = _props => {
  // Setup
  const author = "sebranly";
  const authorAvatar = "https://avatars1.githubusercontent.com/u/25478895";
  const authorUrl = `https://github.com/${author}`;
  const currentYear = new Date(Date.now()).getFullYear();

  const renderLibrary = (library: string) => {
    return (
      <ListItem key={library}>
        <Text>{library}</Text>
      </ListItem>
    );
  };

  const libraries = [
    "@testing-library/react-native",
    "emoji-from-text",
    "native-base",
    "node-emoji",
    "react-native",
    "react-native-easy-grid"
  ].map(v => renderLibrary(v));

  return (
    <Container>
      <Content>
        <List>
          <Separator bordered>
            <Text>Application</Text>
          </Separator>
          <ListItem>
            <Text>{`Created in 2018 - ${currentYear}`}</Text>
          </ListItem>
          <ListItem>
            <Text>{`Version 1.0.0`}</Text>
          </ListItem>
          <ListItem
            onPress={() => {
              Linking.openURL(`${authorUrl}/WordyWorld`);
            }}
          >
            {!IS_TEST && (
              <Left style={styles.flex1}>
                <Icon name="ios-git-network" />
              </Left>
            )}
            <Text style={styles.flex4}>Project's GitHub</Text>
          </ListItem>
          <Separator bordered>
            <Text>Author</Text>
          </Separator>
          <ListItem
            onPress={() => {
              Linking.openURL(authorUrl);
            }}
          >
            <Left style={styles.flex1}>
              <Thumbnail source={{ uri: authorAvatar }} />
            </Left>
            <Body style={styles.flex4}>
              <Text>{author}</Text>
              <Text note>Developer/Tester</Text>
            </Body>
          </ListItem>

          <Separator bordered>
            <Text>Libraries used</Text>
          </Separator>
          {libraries}
        </List>
      </Content>
    </Container>
  );
};

const styles = StyleSheet.create({
  flex1: {
    flex: 1
  },
  flex4: {
    flex: 4
  }
});

export { About };
