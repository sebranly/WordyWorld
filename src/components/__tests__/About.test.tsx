// Vendor
import React from "react";
import { render } from "@testing-library/react-native";

// Internal
import { About } from "../About";

it("renders", () => {
  const props = {};
  const { container } = render(<About {...props} />);
  expect(container.children.length).toBeGreaterThan(0);
  expect(container.children).toMatchSnapshot();
});
