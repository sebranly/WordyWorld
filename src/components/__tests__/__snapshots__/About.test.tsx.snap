// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      }
    }
  >
    <ScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          undefined,
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      viewIsInsideTabBar={false}
    >
      <View
        style={Object {}}
      >
        <View
          bordered={true}
          style={
            Object {
              "backgroundColor": "#F0EFF5",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "borderTopWidth": 0.5,
              "flex": 1,
              "height": 35,
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingTop": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#777",
                "fontFamily": "System",
                "fontSize": 12,
              }
            }
            uppercase={false}
          >
            Application
          </Text>
        </View>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              Created in 2018 - 2020
            </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              Version 1.0.0
            </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "alignSelf": "center",
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                  },
                  Object {
                    "flex": 4,
                  },
                ]
              }
              uppercase={false}
            >
              Project's GitHub
            </Text>
          </View>
        </TouchableHighlight>
        <View
          bordered={true}
          style={
            Object {
              "backgroundColor": "#F0EFF5",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "borderTopWidth": 0.5,
              "flex": 1,
              "height": 35,
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingTop": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#777",
                "fontFamily": "System",
                "fontSize": 12,
              }
            }
            uppercase={false}
          >
            Author
          </Text>
        </View>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                  },
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "uri": "https://avatars1.githubusercontent.com/u/25478895",
                  }
                }
                style={
                  Object {
                    "borderRadius": 28,
                    "height": 56,
                    "width": 56,
                  }
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": null,
                    "alignSelf": null,
                    "flex": 1,
                  },
                  Object {
                    "flex": 4,
                  },
                ]
              }
            >
              <Text
                style={
                  Object {
                    "color": "#000",
                    "fontFamily": "System",
                    "fontSize": 16,
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                uppercase={false}
              >
                sebranly
              </Text>
              <Text
                note={true}
                style={
                  Object {
                    "color": "#808080",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "200",
                    "marginLeft": 10,
                    "marginRight": 10,
                  }
                }
                uppercase={false}
              >
                Developer/Tester
              </Text>
            </View>
          </View>
        </TouchableHighlight>
        <View
          bordered={true}
          style={
            Object {
              "backgroundColor": "#F0EFF5",
              "borderBottomWidth": 0.5,
              "borderColor": "#c9c9c9",
              "borderTopWidth": 0.5,
              "flex": 1,
              "height": 35,
              "justifyContent": "center",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingTop": 12,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#777",
                "fontFamily": "System",
                "fontSize": 12,
              }
            }
            uppercase={false}
          >
            Libraries used
          </Text>
        </View>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              @testing-library/react-native
            </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              emoji-from-text
            </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              native-base
            </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              node-emoji
            </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              react-native
            </Text>
          </View>
        </TouchableHighlight>
        <TouchableHighlight
          activeOpacity={0.85}
          delayPressOut={100}
          underlayColor="#DDD"
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomWidth": 0.5,
                "borderColor": "#c9c9c9",
                "flexDirection": "row",
                "marginLeft": 16,
                "paddingBottom": 13,
                "paddingRight": 16,
                "paddingTop": 13,
              }
            }
          >
            <Text
              style={
                Object {
                  "alignSelf": "center",
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                }
              }
              uppercase={false}
            >
              react-native-easy-grid
            </Text>
          </View>
        </TouchableHighlight>
      </View>
    </ScrollView>
  </View>,
]
`;
